### DISCUSSION TITLE
FAILURE: Build failed with an exception - invalid flag: --release

### DISCUSSION THREAD

**Discussion by Skjolberg** [mapstruct/mapstruct#3301] [Q&A]
Hello!

I'm trying to use MapStruct in my project, when I have everything ready to compile I get the error below, it's quite strange.

As data to take into account:
- My project is multiversion, I have modules compiled in Java 17, others in Java 11, Java 8... but the main module that wraps all is compiled with Java 8, it is a plugin for a game and depending on the server version one module or another will be used.
- I don't use the "--release" flag in the Java Compiler.
- I have this enabled; Build, Execution, Deployment -> Compiler -> Annotation Processors

It only happens when using MapStruct, if I remove it from my project there is no problem. 

If there is any other information I can offer you do not hesitate to ask me.

Thx for your help :)

Gradle version (kotlin): 7.1

Trace: https://pastes.dev/eDpCjOEdbq

Console:

```yaml
21:34:37: Executing 'shadowJar'...

Type-safe dependency accessors is an incubating feature.
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :common:compileJava FAILED
1 actionable task: 1 executed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':common:compileJava'.
> invalid flag: --release

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 293ms
21:34:38: Execution finished 'shadowJar'.

**Comment by filiphr** [comment]
>It only happens when using MapStruct, if I remove it from my project there is no problem.

Are you using some other annotation processors?

To me this looks like a Gradle bug. Have you tried with a different gradle version?

**Comment by Skjolberg** [comment]
Yes, apparently it was a misconfiguration of Gradle. 

Regards.

**Reply by MrCreosote** [reply]
Could you please describe the fix? I'm currently trying to debug the same error on gradle 8.5
